conn.go

```diff
@@ -322,11 +351,15 @@ func (conn *Conn) inWorker() {
 					Name:   iface + "." + member,
 					Body:   msg.Body,
 				}
-				conn.signalsLck.Lock()
-				for _, ch := range conn.signals {
-					ch <- signal
+				conn.signalQueueLck.Lock()
+				if conn.signalQueueWakeUp != nil {
```

Comment:
Ah, there's a race during shutdown between these worker threads and the shared Conn object. Could you add a comment in here to that effect?